consensus/ibft/ibft.go

```diff
@@ -1389,6 +1352,18 @@ func (i *Ibft) Close() error {
 	return nil
 }
 
+// SetHeaderHash updates hash calculation function for IBFT
+func (i *Ibft) SetHeaderHash() {
+	types.HeaderHash = func(h *types.Header) types.Hash {
+		hash, err := i.signer.CalculateHeaderHash(h)
+		if err != nil {
+			return types.ZeroHash
+		}
+
+		return hash
+	}
+}
```

Comment:
Is there any way we can mitigate setting this **global** callback to something specific?

Usually this is a red flag for bad module design, can we evaluate where we need this `types.HeaderHash`?

cc @dbrajovic

Reply:
I know we were setting this global variable previously as well, I'm just wondering if we can go away with this shenanigans 👀

Reply:
HashCalculation is used widely and I know global variable is not proper

Reply:
We need to figure out which modules are using this

Reply:
Given how widespread usage of `Header`'s `ComputeHash()` method is, I'd say this is best we can do right now. 

A good direction to think about this is how many modules would actually need the hash of a given `Header`. If that list isn't too long, then maybe this implementation could be more specific.